docs/pages/release_notes.md

```diff
@@ -1,6 +1,113 @@
 Release Notes {#releasenotes}
 ==============
 
+v3.5.0
+==============
+
+Major Updates
+-------------
+
+* ArrayFire now supports threaded applications
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1706)</sup>
+* Added Canny edge detector
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1743)</sup>
+* Added Sparse-Dense arithmetic operations
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1696)</sup>
+
+Features
+--------
+
+* ArrayFire Threading
+  * \ref af::array can be read by multiple threads
+  * All ArrayFire functions can be executed concurrently by multiple threads.
+  * Threads can operate on different devices to simplify Muli-device workloads
+* New Canny edge detector function, \ref af::canny().
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1743)</sup>
+  * Can automatically calculate high threshold with `AF_CANNY_THRESHOLD_AUTO_OTSU`
+  * Supports both L1 and L2 Norms to calculate gradients
+* New tuned OpenCL BLAS backend,
+  [CLBlast](https://github.com/arrayfire/arrayfire/pull/1727).
+
+Improvements
+------------
+
+* Performance improvements when using \ref af::reorder()
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1766)</sup>
+* Improved unified backend performance
+  <sup>[1](https://github.com/arrayfire/arrayfire/pull/1770)</sup>
+* ArrayFire now depends on Forge
+  v1.0. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1800)</sup>
+* Can now specify the FFT plan cache size using the
+  \ref af::setFFTPlanCacheSize() function.
+* Get the number of physical bytes allocated by the memory manager
+  \ref `af_get_allocated_bytes()`
+* \ref af::dot() can now return a scalar value to the host.
+
+Bug Fixes
+---------
+
+* Fixed improper release of default Mersenne random
+  engine. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1716)</sup>
+* Fixed assignment bug in CPU
+  backend. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1765)</sup>
+* Fixed memory leak in
+  \ref af_fir. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1765)</sup>
+* Fixing correctness of \ref af_pow for complex numbers by using Cartesian
+  form. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1765)</sup>
+* Corrected \ref af::select() with indexing in CUDA and OpenCL
+  backends. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1731)</sup>
+* Workaround for VS2015 compiler ternary
+  bug. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1771)</sup>
+* Fixed memory corruption in
+  `cuda::findPlan()`. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1793)</sup>
+* Argument checks in \ref af_create_sparse_array avoids inputs of type
+  int64. <sup>[1](https://github.com/arrayfire/arrayfire/pull/1747)</sup>
+
+Build fixes
```

Comment:
I guess this https://github.com/arrayfire/arrayfire/pull/1708/files comes under a build fix.